server/crawler/index.ts

```diff
@@ -194,25 +194,52 @@ export default async function getCralwerData(url: string) {
   let author = ''
   let title = ''
   let body = ''
-  let image = ''
+  let ogImage = ''
+  const images:any = []
+
   try {
     const { data: content } = await axios.get(encodeURI(url as string))
+    const { protocol , host} = new URL(url)
     const $ = cheerio.load(content);
     [
       'script',
       'style',
       'svg',
+      'source',
     ].forEach((t: string) => $(t).remove())
     title = $('title').text()
     const metas = $('meta')
-    body = $('body').html() || ''
+    const promiseImg:any = []
 
+    let index = 0
+    const urlIndexObject:any = {}
     const img = $('img')
-    Object.keys(img).forEach((key: any) => {
-      const { src } = img[key].attribs || {};
-      if (!src) {
-        delete img[key]
+    img.each(function() {
+      const src = $(this).attr('src');
+      if (src) {
+        let srcUrl = src
+        if ( !src.startsWith('http') ) {
+          srcUrl = `${protocol}//${host}${src}`
+        }
+        if(!urlIndexObject[srcUrl]){
+          urlIndexObject[srcUrl] = index
+          index += 1
+        }
+        promiseImg.push(axios.get(`${srcUrl}`, {responseType: 'arraybuffer'})
+        .then((element)=> {
+          const elementUrl = element?.config?.url
+          const newFileName = `./img${urlIndexObject[`${elementUrl}`]}`
+          $(this).attr('src', newFileName); 
+          return { element, key: newFileName }
+        })
+        .catch(()=> {}));
       }
+    });
+
+    const imgData:any = await Promise.all(promiseImg)
+    imgData.filter((e: any) => e?.element?.status === 200).map((e: any) => {
+      images.push({ 'data': Buffer.from(e.element.data, 'binary').toString('base64'), 'key': e.key, 'type': e.element.headers['content-type']} );
```

Comment:
Separate the object into multiple lines for readability, also the quotes around keys are not needed